lib/discourse_api/single_sign_on.rb

```diff
@@ -52,11 +59,11 @@ class SingleSignOn
     attr_writer :custom_fields, :sso_secret, :sso_url
 
     def self.sso_secret
-      raise RuntimeError, "sso_secret not implemented on class, be sure to set it on instance"
+      raise ParseError, "sso_secret not implemented on class, be sure to set it on instance"
```

Comment:
Do we need to raise a ParseError here? This seems like a better fit for a `NotImplemented` error.

Reply:
I've changed this to a new `MissingConfigError` exception. `NotImplementedError` is meant to be used when a method isn't supported on the current platform, not to indicate that a method needs implementing, see https://ruby-doc.org/core-2.5.0/NotImplementedError.html.